### DISCUSSION TITLE
Command status output on real time response

### DISCUSSION THREAD

**Discussion by hermanmaleiane** [CrowdStrike/falconpy#487] [Q&A] [answered]
Hi team,

I have some questions regarding a function **check_admin_command_status** on rtr. 
I'm executing a scan on a remote host using this script: Start-MpScan -ScanType QuickScan and i got the cloud_request_id value.

![image](https://user-images.githubusercontent.com/13849459/145235621-30907c31-e660-4471-af46-58a70d7c317e.png)

This values is displayed for all calls with that  **cloud_request_id**. 
The value doesn't change even if the scan is completed. It says complete **true** while host  is scanning and after scan is finished.

![image](https://user-images.githubusercontent.com/13849459/145235821-518f91cb-f115-4a5b-97d6-93d3bfbda163.png)

**Comment by jshcodes** [comment, marked as answer]
Hi @hermanmaleiane -

I believe the question is "Why am I getting a `true` back from RTR when the scan has not completed?".

This is because the complete you are receiving from the request to the RTR API is not an indication of the underlying command's status, just that the command you executed was run and returned the process. (eg. It "completed".) 

This is why this value is `true` both times. 

So for this scenario, you would want to: 
- execute the scan command
- check that the command executed (by checking the value of _complete_ looking for `true`)
- then you would execute the followup "Get-MpComputerStatus" command to retrieve the scan results
- again checking that the command executed by checking the value of _complete_.
- When you get a `true` back, you would return the value of _stdout_ or _stderr_ depending.
    - If you have output in _stderr_ like shown above, the scan has not finished, sleep for N seconds and retry.
